sdk/tables/aztable/connection_string.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package aztable
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+var ErrConnectionString = errors.New("connection string is either blank or malformed")
+
+func NewTableClientFromConnectionString(tableName string, connectionString string, options *TableClientOptions) (*TableClient, error) {
+	endpoint, credential, err := parseConnectionString(connectionString)
+	if err != nil {
+		return nil, err
+	}
+	return NewTableClient(tableName, endpoint, *credential, options)
+}
+
+func NewTableServiceClientFrommConnectionString(connectionString string, options *TableClientOptions) (*TableServiceClient, error) {
+	endpoint, credential, err := parseConnectionString(connectionString)
+	if err != nil {
+		return nil, err
+	}
+	return NewTableServiceClient(endpoint, *credential, options)
+}
+
+func convertConnStrToMap(connStr string) (map[string]string, error) {
+	ret := make(map[string]string)
+	connStr = strings.TrimRight(connStr, ";")
+
+	splitString := strings.Split(connStr, ";")
+	if len(splitString) == 0 {
+		return ret, ErrConnectionString
+	}
+	for _, stringPart := range splitString {
+		parts := strings.Split(stringPart, "=")
```

Comment:
This might not happen in Tables but in Service Bus we can have embedded '='s in the connection string.

My understanding is that's legal so you might want to say .SplitN(stringPart, "=", 2), which will only split _once_.

Reply:
yes the sharedKey typically has "==" in it